src/int/specialized_div_rem/mod.rs

```diff
@@ -0,0 +1,95 @@
+// This `specialized_div_rem` module is from version 0.2.0 of the
+// `specialized-div-rem` crate (the build system cannot handle
+// including it as a regular dependency).
+
+#[macro_use]
+mod binary_long;
+
+#[macro_use]
+mod delegate;
+
+#[cfg(not(target = "x86_64"))]
+#[macro_use]
+mod trifecta;
+
+#[cfg(target = "x86_64")]
+#[macro_use]
+mod asymmetric;
+
+/// This function is unsafe, because if the quotient of `duo` and `div` does not
+/// fit in a `u64`, a floating point exception is thrown.
+#[cfg(target = "x86_64")]
+#[inline]
+unsafe fn u128_by_u64_div_rem(duo: u128, div: u64) -> (u64, u64) {
+    let quo: u64;
+    let rem: u64;
+    let duo_lo = duo as u64;
+    let duo_hi = (duo >> 64) as u64;
+    asm!("divq $4"
```

Comment:
:( My codegen backend cg_clif doesn't support inline asm.

Reply:
We need `divq` in order for x86-64 to reach it full potential

Reply:
LLVM is able to use `mulq` but it does not seem to be able to use `divq`. [playground example](https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=154cc34651b3e2cb2b15cbf928d27400)

Reply:
While I have just patched away other inline asm usages in libcore, that is not possible with `compiler-builtins`, as libstd depends on the crates.io version. It isn't distributed with the rest of the libstd source.

Maybe add a feature flag to disable this? I can then patch libstd to enable it by default.

Reply:
I have added a feature flag